ref/Microsoft.Build.Framework/net/Microsoft.Build.Framework.cs

```diff
@@ -1,5 +1,128 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.Build.Eventing
```

Comment:
I recommend that you merge these EventSources into one EventSource class unless there isn't an assembly you can put it into that is accessible in all the necessary places.  There is cost to having lots of EventSources:

 - Initialization: There is some fixed cost to initialization that you'll pay for every one of these.
 - Usage: Folks have to know about all of these.

If the goal is to be able to enable/disable these events individually, there are better ways to do this through the keywords mechanism.

I also recommend that you use more global names.  Remember that the EventSource name is part of a global namespace of EventSources and so it's important to disambiguate.  You should use something like "Microsoft-Build" for the name of a single EventSource that includes all of these events.

I'm happy to walk through the details on how to structure the EventSource so that it meets your needs while doing things in an EventSource idiomatic way that users will understand.

Reply:
Somehow, I didn't realize that you could have them all in a single class. I can certainly do that—thanks for the suggestion!

We originally thought it might be helpful to separate them out to some extent, but if our main focus is their use in RPS, and we want to capture all information in RPS, I'm not sure we still need that.

The names were temporary until we decided what names we should use, and we never got around to changing them. It will be much simpler when they're all in one class.

Reply:
Excellent.